src/logic/map_objects/tribes/soldier.cc

```diff
@@ -376,7 +376,6 @@ void Soldier::set_level(uint32_t const health,
 	set_evade_level(evade);
 }
 void Soldier::set_health_level(const uint32_t health) {
-	assert(health_level_ <= health);
```

Comment:
Why did you remove the asserts here?

Reply:
Until now this is used only by trainingsites, where soldier levels can only be increased but should not be decreased. But in the editor we also allow the designer to reconsider and turn a hero back into a rookie.

Reply:
Until now this is used only by trainingsites, where soldier levels can only be increased but should not be decreased. But in the editor we also allow the designer to reconsider and turn a hero back into a rookie.

Reply:
Thanks for the explanation